src/plotconfig.jl

```diff
@@ -335,10 +335,11 @@ function config_kwargs!(cfg::PlotConfig; kwargs...)
 end
 
 """
-    update_axis(support_axis::NamedTuple; kwargs...)
-Update values of `NamedTuple{key = value}`\\
+    update_axis!(support_axis::NamedTuple; kwargs...)
+Update values of `NamedTuple{key = value}`.\\
 Used for supportive axes to make users be able to flexibly change them.
 """
 function update_axis!(support_axis::NamedTuple; kwargs...)
-    return (; support_axis..., kwargs...)
+    support_axis = (; support_axis..., kwargs...)
```

Comment:
given that you create a new axis and dont make a in-place modification, this function should drop it's `!` and be called `update_axis`